server/sonar-server/src/main/java/org/sonar/server/issue/index/IssueIteratorForSingleChunk.java

```diff
@@ -0,0 +1,257 @@
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Maps;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Collection;
+import java.util.Date;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+import org.apache.commons.lang.StringUtils;
+import org.sonar.api.resources.Scopes;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.rules.RuleType;
+import org.sonar.db.DatabaseUtils;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.ResultSetIterator;
+
+import static org.sonar.api.utils.DateUtils.longToDate;
+import static org.sonar.db.DatabaseUtils.getLong;
+
+/**
+ * Scrolls over table ISSUES and reads documents to populate
+ * the issues index
+ */
+class IssueIteratorForSingleChunk implements IssueIterator {
+
+  private static final String[] FIELDS = {
+    // column 1
+    "i.kee",
+    "root.uuid",
+    "i.updated_at",
+    "i.assignee",
+    "i.gap",
+    "i.issue_attributes",
+    "i.line",
+    "i.message",
+    "i.resolution",
+    "i.severity",
+
+    // column 11
+    "i.manual_severity",
+    "i.checksum",
+    "i.status",
+    "i.effort",
+    "i.author_login",
+    "i.issue_close_date",
+    "i.issue_creation_date",
+    "i.issue_update_date",
+    "r.plugin_name",
+    "r.plugin_rule_key",
+
+    // column 21
+    "r.language",
+    "p.uuid",
+    "p.module_uuid_path",
+    "p.path",
+    "p.scope",
+    "i.tags",
+    "i.issue_type"
+  };
+
+  private static final String SQL_ALL = "select " + StringUtils.join(FIELDS, ",") + " from issues i " +
+    "inner join rules r on r.id=i.rule_id " +
+    "inner join projects p on p.uuid=i.component_uuid " +
+    "inner join projects root on root.uuid=i.project_uuid";
+
+  private static final String PROJECT_FILTER = " AND root.uuid=?";
+  private static final String ISSUE_KEY_FILTER_PREFIX = " AND i.kee IN (";
+  private static final String ISSUE_KEY_FILTER_SUFFIX = ")";
+
+  static final Splitter TAGS_SPLITTER = Splitter.on(',').trimResults().omitEmptyStrings();
+  static final Splitter MODULE_PATH_SPLITTER = Splitter.on('.').trimResults().omitEmptyStrings();
+
+  private final DbSession session;
+
+  @CheckForNull
+  private final String projectUuid;
+
+  @CheckForNull
+  private final Collection<String> issueKeys;
+
+  private final PreparedStatement stmt;
+  private final ResultSetIterator<IssueDoc> iterator;
+
+  IssueIteratorForSingleChunk(DbClient dbClient, @Nullable String projectUuid, @Nullable Collection<String> issueKeys) {
+    this.projectUuid = projectUuid;
+    this.issueKeys = issueKeys;
+    this.session = dbClient.openSession(false);
+    try {
+
+      String sql = createSql();
+      stmt = dbClient.getMyBatis().newScrollingSelectStatement(session, sql);
+      try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1141)